### DISCUSSION TITLE
support custom APIs

Thank you, developers. Could you please support custom APIs?

### DISCUSSION THREAD

**Discussion by marcusbai** [logancyang/obsidian-copilot#552] [Q&A]
Thank you, developers. Could you please support custom APIs?

**Comment by logancyang** [comment]
You can use OpenAI proxy in advanced settings.

**Comment by marcusbai** [comment]
> You can use OpenAI proxy in advanced settings.

I believe I've come up with another workaround. I've tweaked the main.js file to replace the relevant instances of [openai.com](http://openai.com/) with a custom URL, and it's working like a charm so far. The only drawback is that I'll need to manually update it after each new release. Nonetheless, I really appreciate your help.

**Comment by logancyang** [comment]
@marcusbai no need to do that manually, the openai proxy base url in adv setting does it for you.

**Comment by marcusbai** [comment]
> @marcusbai no need to do that manually, the openai proxy base url in adv setting does it for you.

thank you
